### PR TITLE
Add install all packages scenario

### PR DESCRIPTION
**Pull request by asmorodskyi**
1. Create generic function in utils which allow to install all packages
from target repository
2. Create module which use this function
3. Use it for testing HPC repository
Done in terms of poo#18370

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -891,6 +892,29 @@ sub service_action {
...
+    if (get_var('INSTALL_ALL_EXCEPT')) {
+        #spliting space separated list of packages into array to iterate over it
+        my @packages_array = split(/ /, get_var('INSTALL_ALL_EXCEPT'));
+        my $grep_regex = "";
+        #loop which will build regex from package names array in form "package1$|package2$|package3$" expected by grep -vE
+        foreach my $package (@packages_array) {
+            if (length $grep_regex) { $grep_regex .= "|" };
+            $grep_regex .= sprintf("$package\$");
+        }
+        $grep_str = sprintf("|grep -vE \"(%s)\"", $grep_regex);
```
Comment (okurz) [inline lib/utils.pm]: maybe use "join" to join the package entries together

File: tests/hpc/hpc_init.pm [hunk 2]
```diff
@@ -18,6 +18,8 @@ use utils;
 use lockapi;
 
 sub run() {
+    # current structure of HPC tests in sle/main.pm make it easiest way to skip this module for HPC=repository
+    if(check_var('HPC','repository')) { return; }
```
Comment (okurz) [inline tests/hpc/hpc_init.pm]: `return if check_var('HPC', 'repository');`

File: lib/utils.pm (hunk 1 above)
Comment (asmorodskyi) [inline lib/utils.pm]: join will not work here , my goal is to get 
"package1$|package2$|package3$" I am adding "$|" after first two and "$" after last one , join can add only same things , right ?

File: tests/hpc/hpc_init.pm (hunk 2 above)
Comment (asmorodskyi) [inline tests/hpc/hpc_init.pm]: Done

File: lib/utils.pm (hunk 1 above)
Comment (okurz) [inline lib/utils.pm]: But you don't need the `$` after every word. Probably you want just

```perl
my @excludes = split / /, get_var('INSTALL_ALL_EXCEPT');
my $exclude_re = '(' . join('|', @excludes) . ')$';
$grep_str = "| grep -vE \"$exclude_re\"";
```

File: lib/utils.pm [hunk 3]
```diff
@@ -891,6 +892,29 @@ sub service_action {
     }
 }
 
+=head2 install_all_from_repo
+   will install all packages in repo defined by INSTALL_ALL_REPO variable
+   with ability to exclude some of them by using INSTALL_ALL_EXCEPT which suppose to contain
+   space separated list of packages
```
Comment (asmorodskyi) [inline lib/utils.pm]: Done.

P.S. Can we instruct travis to check this ?

Comment (okurz) [inline lib/utils.pm]: yes, https://github.com/os-autoinst/os-autoinst/blob/master/t/05-pod.t might be a start.

**Comment by asmorodskyi** [comment]
actually I need '$' after every word , because I don't want to filter out "package1-devel" when I specified "package1" in a list

File: lib/utils.pm (hunk 1 above)
Comment (asmorodskyi) [inline lib/utils.pm]: actually I need '$' after every word , because I don't want to filter out "package1-devel" when I specified "package1" in a list

Comment (okurz) [inline lib/utils.pm]: I am not sure what you want there. I guess my approach still works but if you really want to append '$' to every token then you can still use

```perl
my $exclude_re = '(' . join('$|', @excludes) . '$)';
```

Comment (asmorodskyi) [inline lib/utils.pm]: I simplify it even more

**Comment by asmorodskyi** [comment]
LGTM by @okurz

**Pull request merged**